atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java

```diff
@@ -166,6 +175,132 @@ public void checkAgainstAccidentalTableV2APIChanges() throws IOException {
+        schema.addCleanupTask(TEST_TABLE_NAME, CLEANUP_TASK_SUPPLIER);
+        assertCleanupRequirementOnTestTableRef(schema, CleanupRequirement.ARBITRARY_ASYNC);
+    }
+
+    @Test
+    public void indexesWithoutCleanupTasksHaveNotNeededCleanupRequirement() {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        schema.addTableDefinition(TEST_TABLE_NAME, getSimpleTableDefinition(TABLE_REF));
+
+        IndexDefinition indexDefinition = new IndexDefinition(IndexDefinition.IndexType.ADDITIVE);
+        indexDefinition.onTable(TEST_TABLE_NAME);
+        schema.addIndexDefinition(TEST_TABLE_NAME, indexDefinition);
+
+        assertThat(getCleanupRequirements(schema).get(
+                TableReference.create(Namespace.EMPTY_NAMESPACE, TEST_INDEX_NAME)))
+                .isEqualTo(CleanupRequirement.NOT_NEEDED);
+    }
+
+    @Test
+    public void indexWithOneCleanupTaskHasArbitraryAsyncCleanupRequirement() {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        schema.addTableDefinition(TEST_TABLE_NAME, getSimpleTableDefinition(TABLE_REF));
+
+        IndexDefinition indexDefinition = new IndexDefinition(IndexDefinition.IndexType.ADDITIVE);
+        indexDefinition.onTable(TEST_TABLE_NAME);
+        schema.addIndexDefinition(TEST_TABLE_NAME, indexDefinition);
+
+        schema.addCleanupTask(TEST_INDEX_NAME, CLEANUP_TASK_SUPPLIER);
+
+        assertThat(getCleanupRequirements(schema).get(
+                TableReference.create(Namespace.EMPTY_NAMESPACE, TEST_TABLE_NAME)))
+                .isEqualTo(CleanupRequirement.NOT_NEEDED);
+        assertThat(getCleanupRequirements(schema).get(
+                TableReference.create(Namespace.EMPTY_NAMESPACE, TEST_INDEX_NAME)))
+                .isEqualTo(CleanupRequirement.ARBITRARY_ASYNC);
+    }
+
+    @Test
+    public void tableWithMultipleCleanupTasksHasArbitraryAsyncCleanupRequirement() {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        schema.addTableDefinition(TEST_TABLE_NAME, getSimpleTableDefinition(TABLE_REF));
+        IntStream.range(0, 10)
+                .forEach(unused -> schema.addCleanupTask(TEST_TABLE_NAME, CLEANUP_TASK_SUPPLIER));
+        assertCleanupRequirementOnTestTableRef(schema, CleanupRequirement.ARBITRARY_ASYNC);
+    }
+
+    @Test
+    public void streamStoreTablesCreatedWithCorrectCleanupRequirements() {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        String shortName = "f";
+        String longName = "floccinaucinihilipilification";
+
+        setupStreamStore(schema, shortName, longName);
+        Map<TableReference, CleanupRequirement> cleanupRequirements = getCleanupRequirements(schema);
+
+        // TODO (jkong): Change if/when we decide to special-case stream stores
+        assertThat(cleanupRequirements.get(
```

Comment:
extract an assert method that takes a table name and a CleanupRequirement

Reply:
New design has `assertCleanupMetadataInSchemaSatisfies(Schema, String, Consumer<CleanupMetadata>)` that works much like the AssertJ satisfies for a provided schema and string table-name.